src/datasource.ts

```diff
@@ -506,45 +506,74 @@ export class OpenSearchDatasource extends DataSourceApi<OpenSearchQuery, OpenSea
     // it as an integer not as string with digits. This is because elastic will convert the string only if the time
     // field is specified as type date (which probably should) but can also be specified as integer (millisecond epoch)
     // and then sending string will error out.
-    payload = payload.replace(/"\$timeFrom"/g, options.range.from.valueOf().toString());
-    payload = payload.replace(/"\$timeTo"/g, options.range.to.valueOf().toString());
-    payload = getTemplateSrv().replace(payload, options.scopedVars);
-
-    return from(this.post(this.getMultiSearchUrl(), payload)).pipe(
-      map((res: any) => {
-        const er = new OpenSearchResponse(targets, res);
-
-        if (targets.some(target => target.isLogsQuery)) {
-          const response = er.getLogs(this.logMessageField, this.logLevelField);
-          for (const dataFrame of response.data) {
-            enhanceDataFrame(dataFrame, this.dataLinks);
-          }
-          return response;
-        }
+    const createQuery = ts => {
+      let payload = '';
 
-        // TODO: what if only one of the targets is a trace query? Is that possible?
-        // we seems to have the same/similar problem above with logs
-        if (targets.every(target => target.luceneQueryType === LuceneQueryType.Traces)) {
-          const luceneQueryString = targets[0].query;
-          if (getTraceIdFromLuceneQueryString(luceneQueryString)) {
-            return createTraceDataFrame(targets, res.responses[0].hits.hits);
-          }
-          return createListTracesDataFrame(targets, res.responses, this.uid, this.name, this.type);
-        }
+      for (const target of ts) {
+        payload += this.createLuceneQuery(target, options);
+      }
 
-        return er.getTimeSeries();
-      })
+      // We replace the range here for actual values. We need to replace it together with enclosing "" so that we replace
+      // it as an integer not as string with digits. This is because elastic will convert the string only if the time
+      // field is specified as type date (which probably should) but can also be specified as integer (millisecond epoch)
+      // and then sending string will error out.
+      payload = payload.replace(/"\$timeFrom"/g, options.range.from.valueOf().toString());
+      payload = payload.replace(/"\$timeTo"/g, options.range.to.valueOf().toString());
+      payload = getTemplateSrv().replace(payload, options.scopedVars);
+
+      return payload;
+    };
+
+    const traceListTargets = targets.filter(
+      target => target.luceneQueryType === LuceneQueryType.Traces && !getTraceIdFromLuceneQueryString(target.query)
+    );
+    const traceTargets = targets.filter(
+      target => target.luceneQueryType === LuceneQueryType.Traces && getTraceIdFromLuceneQueryString(target.query)
     );
+
+    const otherTargets = targets.filter(target => target.luceneQueryType !== LuceneQueryType.Traces);
+
+    const traceListObs$ =
```

Comment:
Good question, usually "$"s are added to distinguish regular variables from observables. But looking at these again, I renamed them from e.g. traceListObs$ to traceList$, since the names contained a redundancy ("obs" + "$")